### PR TITLE
Adding SARIF V1 read capability to compiler telemetry

### PR DESCRIPTION
**Pull request by marmegh**
Many customers have a strong dependency on SARIF V1 and are therefore unable to utilize the full compiler telemetry functionality. This change addresses this by adding capacity to distinguish SARIF version and convert before logging telemetry summary if needed. Additionally, test coverage was added to verify the reading and telemetry logging in this situation.

### CODE REVIEW

File: src/Test.UnitTests.BinSkim.Driver/CompilerDataLoggerTests.cs [hunk 1]
```diff
@@ -106,10 +109,39 @@ public void CompilerDataLogger_Constructor_ShouldThrowException_WhenTelemetryIsE
...
+            fileSystem
+                .Setup(f => f.FileOpenRead(It.IsAny<string>()))
+                .Returns(new MemoryStream(byteArray));
+
+            List<ITelemetry> sendItems = TestSetup(sarifLogPath, context, Sarif.SarifVersion.OneZeroZero, out CompilerDataLogger compilerDataLogger, fileSystem.Object);
+            compilerDataLogger.Dispose();
+
+            fileSystem.Verify(fileSystem => fileSystem.FileOpenRead(sarifLogPath), Times.Once);
+            sendItems.Count.Should().Be(1);
+        }
```
Comment (marmegh) [inline src/Test.UnitTests.BinSkim.Driver/CompilerDataLoggerTests.cs:142]: This is the added test.
Used the `Native_x86_VS2019_SDL_Enabled.exe` to generate a SARIF v1 output (in addition to the SARIF V2 used in the baseline tests).
Mocked the fileSystem to return this sarif output and verified that the fileSystem was called exactly one time as expected.
Finally, verified that we generated the single telemetry event as expected when calling `CompilerDataLogger.Disepose()`.

File: src/BinSkim.Sdk/CompilerDataLogger.cs [hunk 2]
```diff
@@ -366,6 +373,33 @@ private void WriteSummaryData()
...
+                    actualLog = serializer.Deserialize<SarifLogVersionOne>(reader);
+                }
+
+                var visitor = new SarifVersionOneToCurrentVisitor();
+                visitor.VisitSarifLogVersionOne(actualLog);
+                sarifLog = visitor.SarifLog;
+            }
+
+            return sarifLog;
+        }
```
Comment (marmegh) [inline src/BinSkim.Sdk/CompilerDataLogger.cs:401]: This is the bulk of the code change, loading based on SARIF version.

File: src/ReleaseHistory.md [hunk 3]
```diff
@@ -5,6 +5,7 @@
 * FEATURE: Add new PE `CV_CFL_LANG` language code for `ALIASOBJ` and `Rust`. [530](https://github.com/microsoft/binskim/pull/530)
 * BUGFIX: Fix `BA2014.DoNotDisableStackProtectionForFunctions` to eliminate false positive reports that `GsDriverEntry` has disabled the stack protector. [551](https://github.com/microsoft/binskim/pull/551)
 * BREAKING: Rename `BA2026.EnableAdditionalSdlSecurityChecks` to `BA2026.EnableMicrosoftCompilerSdlSwitch` to clarify rule purpose. [#586](https://github.com/microsoft/binskim/pull/586)
+* BUGFIX: Fix inability to emit `SummaryEvent` when SARIF V1. [613](https://github.com/microsoft/binskim/pull/613)
```
Comment (michaelcfanning) [inline src/ReleaseHistory.md:8]: >Fix [](http://example.com/codeflow?start=10&length=3)

Make this release note searchable based on what customers see when this crashes.

**Review by michaelcfanning — APPROVED** [review]
:shipit:

**Pull request merged**